examples/java/src/main/java/glide/examples/StandaloneExample.java

```diff
@@ -0,0 +1,125 @@
+                        //.useTLS(true);
+                        .build();
+
+        GlideClient client = GlideClient.createClient(config).get();
+        return client;
+    }
+
+    /**
+     * Executes the main logic of the application, performing basic operations such as SET, GET, and
+     * PING using the provided GlideClient.
+     *
+     * @param client An instance of <code>GlideClient</code>.
+     */
+    public static void appLogic(GlideClient client) throws ExecutionException, InterruptedException {
+
+        // Send SET and GET
+        CompletableFuture<String> setResponse = client.set("foo", "bar");
+        Logger.log(Logger.Level.INFO, "app", "Set response is " + setResponse.get());
+
+        CompletableFuture<String> getResponse = client.get("foo");
+        Logger.log(Logger.Level.INFO, "app", "Get response is " + getResponse.get());
+
+        // Send PING to the primary node
+        CompletableFuture<String> pong = client.ping();
+        Logger.log(Logger.Level.INFO, "app", "Ping response is " + pong.get());
+    }
+
+    /** Executes the application logic with exception handling. */
+    private static void execAppLogic() {
+
+        while (true) {
+            GlideClient client = null;
+            try {
+                client = createClient();
+                appLogic(client);
+
+            } catch (CancellationException e) {
+                Logger.log(Logger.Level.ERROR, "glide", "Request cancelled: " + e.getMessage());
+                throw e;
+            } catch (InterruptedException e) {
+                Logger.log(Logger.Level.ERROR, "glide", "Client creation interrupted: " + e.getMessage());
+                Thread.currentThread().interrupt(); // Restore interrupt status
+                throw new CancellationException("Interrupted while creating client");
+            } catch (ExecutionException e) {
+                // All Glide errors will be handled as ExecutionException
+                if (e.getCause() instanceof ClosingException) {
+                    // If the error message contains "NOAUTH", raise the exception
+                    // because it indicates a critical authentication issue.
+                    if (e.getMessage().contains("NOAUTH")) {
+                        Logger.log(
+                            Logger.Level.ERROR, "glide", "Authentication error encountered: " + e.getMessage());
+                    } else {
+                        Logger.log(
+                            Logger.Level.WARN,
+                            "glide",
+                            "Client has closed and needs to be re-created: " + e.getMessage());
+                    }
+                } else if (e.getCause() instanceof ConnectionException) {
+                    // The client wasn't able to reestablish the connection within the given retries
+                    Logger.log(Logger.Level.ERROR, "glide", "Connection error encountered: " + e.getMessage());
```

Comment:
I wonder if we should throw here since the connection couldn't be established within the retries, or if we should just continue to the next iteration of the while loop like we currently have